acme_diags/driver/annual_cycle_zonal_mean_driver.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import print_function
+
+from typing import TYPE_CHECKING, Any, Dict, List
+
+import cdutil
+import MV2
+from cdms2 import createAxis
+from cdms2.axis import TransientAxis
+
+from acme_diags.driver.utils.dataset import Dataset
+from acme_diags.driver.utils.general import (
+    get_name_and_yrs,
+    regrid_to_lower_res,
+    save_ncfiles,
+)
+from acme_diags.plot import plot
+
+if TYPE_CHECKING:
+    from cdms2.tvariable import TransientVariable
+
+    from acme_diags.parameter.core_parameter import CoreParameter
+
+
+def run_diag(parameter: "CoreParameter"):
```

Comment:
Moved `run_diag` to the top of the file and added docstring and type annotations.